File: src/Algorithms/Mergesort.py
from random import randint

from src.HelperFunctions.merge import merge


def merge_sort(numbers:list[int],left:int,right:int)->list[int]:
    """
    Sorts and returns the given list of integers using mergesort.
    """
    if left < right:
        mid = (left+right) // 2
        merge_sort(numbers,left,mid)
        merge_sort(numbers,mid+1,right)
        merge(numbers,left,mid,right)
    return numbers
if __name__ == "__main__":
    length = int(input("How many elements should each list contain?"))
    arr = [0] * length
    for j in range(length):
        arr[j] = randint(0,1000000)
    merge_sort(arr,0,len(arr)-1)
    print(arr)

File: src/Algorithms/mergesort.c
#include "merge.h"
#include <stdlib.h>
#include <time.h>
#include "stdio.h"

void mergesort(int array[],int left,int right){
    int mid;
    if (left<right){
        mid = (left+right) / 2;
        mergesort(array,left,mid);
        mergesort(array,mid+1,right);
        merge(array,left,mid,right);
    }
}


int main(){
    srand((unsigned)time(NULL));
    int indices = 100;
    int array[indices];
    for (int i=0;i<indices;i++){
        array[i] = rand() % 101;
    }
    mergesort(array,0, indices);
    for (int i = 0;i<indices;i++){
        printf("%d ,",array[i]);
    }
}
